Tests/SentryTests/SentryClientTests.swift

```diff
@@ -73,17 +87,43 @@ class SentryClientTest: XCTestCase {
         }
     }
     
-    func testCaptureMessageWithoutStackrace() {
-        let eventId = fixture.getSut().capture(message: message, scope: nil)
-
+    func testCaptureEvent() {
+        let event = Event(level: SentryLevel.fatal)
+        event.message = message
+        let scope = Scope()
+        let expectedTags = ["tagKey": "tagValue"]
+        scope.setTags(expectedTags)
+        
+        let eventId = fixture.getSut().capture(event: event, scope: scope)
+        
         XCTAssertNotNil(eventId)
-
         fixture.assertLastSentEvent { actual in
-            XCTAssertEqual(SentryLevel.info, actual.level)
-            XCTAssertEqual(message, actual.message)
+            XCTAssertEqual(event.level, actual.level)
+            XCTAssertEqual(event.message, actual.message)
             XCTAssertNil(actual.debugMeta)
             XCTAssertNil(actual.threads)
-            XCTAssertNotNil(actual.dist)
+            
+            if let actualTags = actual.tags {
+                XCTAssertEqual(expectedTags, actualTags)
+            } else {
+                XCTFail("Tags of scope not applied to event.")
+            }
+        }
+    }
+    
+    func testCaptureEventWithStacktrace() {
+        let event = Event(level: SentryLevel.fatal)
+        event.message = message
+        let eventId = fixture.getSut(configureOptions: { options in
```

Comment:
M: can you somehow reuse `testCaptureMessageWithStacktrace`? maybe adding the event as a return type

Reply:
Good point 👍. We now compare `SentryDebugMeta`, but I had to implement `isEqual` and `hash` for this. In order to compare the threads I would also need to implement this for `SentryThread`, `SentryStacktrace`, and `SentryFrame`, which is quite some work. I would like to do this, but not in this PR.

Reply:
I'm fine doing it in another PR, but I feel that `SentryStacktrace` and `SentryFrame` at least are a vital part of this feature, should we not consider asserting them in this very PR?

Reply:
Yes, they are vital parts of this feature. We already know that they are working fine and we didn't have any tests before. this PR already improves the current tests. I want to do it, but this is going to blow up the PR a lot.